var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -420,8 +421,21 @@ def write_rpath_specs(self):
         set_install_permissions(specs_file)
 
     def setup_environment(self, spack_env, run_env):
-        run_env.set('CC', join_path(self.spec.prefix.bin, 'gcc'))
-        run_env.set('CXX', join_path(self.spec.prefix.bin, 'g++'))
-        run_env.set('FC', join_path(self.spec.prefix.bin, 'gfortran'))
-        run_env.set('F77', join_path(self.spec.prefix.bin, 'gfortran'))
-        run_env.set('F90', join_path(self.spec.prefix.bin, 'gfortran'))
+        # Search prefix directory for possibly modified compiler names
+        from spack.compilers.gcc import Gcc as GccCompiler
+        compiler_names = {}
+        bin_path = self.spec.prefix.bin
+        for name in os.listdir(bin_path):
+            path = os.path.join(bin_path, name)
+            if os.path.isfile(path) and not os.path.islink(path):
+                for suffix in GccCompiler.suffixes:
+                    name = re.sub(suffix, "", name)
```

Comment:
@sethrj 

This appears to work but IMO requires more examination than is necessary (e.g. while `'gcc'` could be overwritten as a key multiple times in this loop I imagine the expectation is that there is only one such gcc file matching the suffix criteria). I think logic like the following would be more straightforward:

```
bin_names = os.listdir(bin_path)
for exe_name in ['gcc', 'g++',...]:
  for suffix in GccCompiler.suffixes:
    alt_names = list(x for x in bin_names if re.match(exe_name + suffix, x))
    if alt_names:
      compiler_names[exe_name] = alt_names[0]
      break
```
But if you find my suggestion to be less clear (or see a problem with it) then this could remain the same but needs more explanation in a `#`-style comment (e.g. explaining what I mentioned above).

Reply:
True, since I'm *only* using gcc/g++/gfortran it would be fine to explicitly loop over them. However, I think some installations (MacPorts for example) will have both `gcc` and `gcc-mp-9`, where one of them is a symlink. I also realized from your suggestion that the suffixes don't have `$` at the end of them so there could be a false match in my original method. I'm committing a new version that is more general and looks like `spack.compilers.arguments_to_detect_version_fn`. Let me know if this looks better.

Based on your suggestion